src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -57,90 +55,90 @@ const defaultProps = {
 };
 
 function NewContactMethodPage(props) {
-    const [login, setLogin] = useState('');
-    const [password, setPassword] = useState('');
     const loginInputRef = useRef(null);
 
-    const handleLoginChange = useCallback((value) => {
-        setLogin(value.trim());
-    }, []);
+    const isFormValid = (values) => {
+        const phoneLogin = !_.isEmpty(values.phoneOrEmail) ? LoginUtils.appendCountryCode(LoginUtils.getPhoneNumberWithoutSpecialChars(values.phoneOrEmail)) : '';
 
-    const handlePasswordChange = useCallback((value) => {
-        setPassword(value.trim());
-    }, []);
+        const errors = {};
 
-    const isFormValid = useMemo(() => {
-        const phoneLogin = LoginUtils.appendCountryCode(LoginUtils.getPhoneNumberWithoutSpecialChars(login));
+        if (_.isEmpty(values.phoneOrEmail)) {
+            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', props.translate('contacts.genericFailureMessages.contactMethodRequired'));
+        }
 
-        return (Permissions.canUsePasswordlessLogins(props.betas) || password)
-            && (Str.isValidEmail(login) || parsePhoneNumber(phoneLogin).possible);
-    }, [login, password, props.betas]);
+        if (!_.isEmpty(values.phoneOrEmail) && !(parsePhoneNumber(phoneLogin).possible || Str.isValidEmail(values.phoneOrEmail))) {
+            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', props.translate('contacts.genericFailureMessages.invalidContactMethod'));
+        }
 
-    const submitForm = useCallback(() => {
-        const phoneLogin = LoginUtils.appendCountryCode(LoginUtils.getPhoneNumberWithoutSpecialChars(login));
-        const parsedPhoneNumber = parsePhoneNumber(phoneLogin);
-        const userLogin = parsedPhoneNumber.possible ? parsedPhoneNumber.number.e164 : login;
+        if (!_.isEmpty(values.phoneOrEmail) && lodashGet(props.loginList, (values.phoneOrEmail).toLowerCase())) {
+            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', props.translate('contacts.genericFailureMessages.enteredMethodIsAlreadySubmited'));
+        }
 
-        // If this login already exists, just go back.
-        if (lodashGet(props.loginList, userLogin)) {
-            Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS);
-            return;
+        if (!Permissions.canUsePasswordlessLogins(props.betas) && _.isEmpty(values.password)) {
+            errors.password = props.translate('contacts.genericFailureMessages.passwordRequired');
         }
 
-        User.addNewContactMethodAndNavigate(userLogin, password);
-    }, [login, props.loginList, password]);
+        return errors;
+    };
+
+    const submitForm = (values) => {
+        const phoneLogin = !_.isEmpty(values.phoneOrEmail) ? LoginUtils.appendCountryCode(LoginUtils.getPhoneNumberWithoutSpecialChars(values.phoneOrEmail)) : '';
```

Comment:
We don't need to check if the `phoneOrEmail` is empty; it's already validated.

```suggestion
        const phoneLogin = LoginUtils.appendCountryCode(LoginUtils.getPhoneNumberWithoutSpecialChars(values.phoneOrEmail));
```

The `parsedPhoneNumber` will have the flag to determine if the `phoneOrEmail` is a phone number or not.